src/Sulu/Bundle/WebsiteBundle/Twig/Seo/SeoTwigExtension.php

```diff
@@ -168,16 +168,22 @@ private function renderAlternateLinks(array $urls, $webspaceKey)
     {
         $html = '';
 
-        $portal = $this->requestAnalyzer->getPortal();
         $defaultLocale = null;
+        $portal = $this->requestAnalyzer->getPortal();
         if ($portal) {
-            $defaultLocale = $portal->getDefaultLocalization()->getLocalization();
-
-            $html .= $this->renderAlternateLink($urls[$defaultLocale], $webspaceKey, $defaultLocale, true);
+            $defaultLocale = $portal->getXDefaultLocalization()->getLocalization();
         }
 
         foreach ($urls as $locale => $url) {
-            $html .= $this->renderAlternateLink($url, $webspaceKey, $locale);
+            // url = '/' means that there is no translation for this page
+            // the only exception is the homepage where the requested resource-locator is false
+            if ($url !== '/' || $this->requestAnalyzer->getResourceLocator() === false) {
```

Comment:
Thinking of it, it would be nice if we wouldn't have to pass these arguments to the twig extension at all, but load them using injected services. Didn't think of that, when I refactored this part... Would be another BC break, so maybe a change for 2.0? /cc @chirimoya 

So I would say this is fine for now.